clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1218,9 +1218,14 @@ void Clang::AddPreprocessingOptions(Compilation &C, const JobAction &JA,
     getToolChain().AddCudaIncludeArgs(Args, CmdArgs);
 
   if (JA.isOffloading(Action::OFK_SYCL) ||
-      Args.hasArg(options::OPT_fsycl_device_only))
+      Args.hasArg(options::OPT_fsycl_device_only)) {
     toolchains::SYCLToolChain::AddSYCLIncludeArgs(D, Args, CmdArgs);
 
+    if (getToolChain().getTriple().isSPIR()) {
```

Comment:
Just to make sure, would:
```suggestion
    if (JA.isDeviceOffloading()) {
```
be unacceptable because DPC++ NVPTX compilations would also pass the condition?